app/javascript/packs/pw-strength.js

```diff
@@ -96,17 +96,35 @@ function disableSubmit(submitEl, length = 0, score = 0) {
   }
 }
 
+/**
+ * @param {HTMLElement?} element
+ *
+ * @return {string[]}
+ */
+export function getForbiddenPasswords(element) {
+  try {
+    return JSON.parse(element.dataset.forbidden);
+  } catch {
+    return [];
+  }
+}
+
 function analyzePw() {
   const { userAgent } = window.navigator;
   const input = document.querySelector(
-    '#password_form_password, #reset_password_form_password, #update_user_password_form_password',
+    [
+      '#password_form_password',
+      '#password_reset_from_disavowal_form_password',
+      '#reset_password_form_password',
+      '#update_user_password_form_password',
+    ].join(','),
```

Comment:
Now that we have the partial reliably setting `data-forbidden` what if we used that to query for instead of these specific IDs?

Reply:
Or maybe we add a specific attribute like `data-password-strength`?

Reply:
The partial is separate from the inputs targeted by these selectors. Right now, the partial only represents the feedback text for password strength. The input itself is rendered outside the partial in the form.

I do think it would probably make most sense to consider the password field with associated strength feedback as a single unit, which would make what you propose a bit more obvious.